cfn-resources/teams/cmd/resource/team-user/team_user.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package teamuser
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231001001/admin"
+)
+
+type TeamUsersAPI interface {
+	GetUserByUsername(ctx context.Context, userName string) (*atlasv2.CloudAppUser, *http.Response, error)
+	AddTeamUser(ctx context.Context, orgID string, teamID string, addUserToTeam *[]atlasv2.AddUserToTeam) (*atlasv2.PaginatedApiAppUser, *http.Response, error)
+	RemoveTeamUser(ctx context.Context, orgID string, teamID string, userID string) (*http.Response, error)
+}
+
+type TeamUsersAPIService struct {
+	mongoDBCloudUsersAPI atlasv2.MongoDBCloudUsersApi
+	teamsAPI             atlasv2.TeamsApi
+}
+
+func NewTeamUsersAPIService(client *atlasv2.APIClient) *TeamUsersAPIService {
+	return &TeamUsersAPIService{
+		mongoDBCloudUsersAPI: client.MongoDBCloudUsersApi,
+		teamsAPI:             client.TeamsApi,
+	}
+}
+
+func (s *TeamUsersAPIService) GetUserByUsername(ctx context.Context, userName string) (*atlasv2.CloudAppUser, *http.Response, error) {
+	return s.mongoDBCloudUsersAPI.GetUserByUsername(context.Background(), userName).Execute()
+}
+
+func (s *TeamUsersAPIService) AddTeamUser(ctx context.Context, orgID string, teamID string, addUserToTeam *[]atlasv2.AddUserToTeam) (*atlasv2.PaginatedApiAppUser, *http.Response, error) {
+	return s.teamsAPI.AddTeamUser(ctx, orgID, teamID, addUserToTeam).Execute()
+}
+
+func (s *TeamUsersAPIService) RemoveTeamUser(ctx context.Context, orgID string, teamID string, userID string) (*http.Response, error) {
+	return s.teamsAPI.RemoveTeamUser(ctx, orgID, teamID, userID).Execute()
+}
+
+func FilterOnlyValidUsernames(mongoDBCloudUsersAPIClient TeamUsersAPI, usernames []string) ([]atlasv2.CloudAppUser, *http.Response, error) {
```

Comment:
Since the intention of team_user and team_user_test is to define and test the business logic in `FilterOnlyValidUsernames`, `GetUsersToAddAndRemove`, and `UpdateTeamUsers` functions, the definition of `NewTeamUsersAPIService` with its default implementation could be defined in separate file `team_user_service`. The mock file could also be renamed to `team_user_service` to make more implicit what is being mocked.

Reply:
Done! I moved it to `team_user_service.go`. The other thing I did is to move the `mocks` folder close to the `team_user_service.go` file. So I just called it mocks. I am not sure if there are strong objections here though, but I think it could be a starting point. I felt that initiatlly I put it under `test` folder and it was a bit far from its real usage.

Reply:
Resolving this conversation but please reopen if you don't agree with my choices.